var_types.py

```diff
@@ -22,61 +26,76 @@ def declare(directory):
-    mantissa_values = [2 ** -i for i in range(1,24)]
-    exponent_values = [2 ** i for i in range(7,-1,-1)]
-    bits = list(map(Directory.directory_to_bit, value_dir.get_children_paths()))
-    exponent = sum([b * e for b, e in zip(exponent_values, bits[1:8])]) - 127
-    mantissa = 1 + sum([b * e for b, e in zip(mantissa_values, bits[9:])])
-    sign = (1, -1)[bits[0]]
-    var_name = parse_string_value(name_directory)
-    value = sign*mantissa*math.pow(2,exponent)
-    attach_variable(var_name,value)
+    value = parse_declare_value(value_dir, Types.float, 32)
+    if value.__class__ is not float:
+        raise ValueError("Type mismatch: expected {0} got {1}".format(float, value.__class__))
+    var_name = parse_string_value([name_directory])
+    attach_variable(var_name, value)
     print(f"VAR TYPE FLOAT {var_name} = {value}")
 
 
 def declare_char(char_dir, name_directory):
-    if char_dir.dirlen() != 8:
-        raise ValueError(
-            f"First subdirectory of directory {char_dir.parent_path.path} of type 'declare char' must have 8 subdirectories "
-            f"ASCII (0-127) but has {char_dir.dirlen()}")
-    char = chr(parse_integer_value(char_dir, False))
-    var_name = parse_string_value(name_directory)
-    attach_variable(var_name, char)
-    print(f"VAR TYPE CHAR {var_name} = {char}")
+    value = parse_declare_value(char_dir, Types.char, 8)
+    if value.__class__ is not str and value.len() > 1:
+        raise ValueError("Type mismatch: expected {0} got {1}".format("char", value.__class__))
+    var_name = parse_string_value([name_directory])
+    attach_variable(var_name, value)
+    print(f"VAR TYPE CHAR {var_name} = {value}")
 
 
 def declare_string(chars_dir, name_directory):
-    string = parse_string_value(chars_dir)
-    var_name = parse_string_value(name_directory)
-    attach_variable(var_name, string)
-    print(f"VAR TYPE STRING {var_name} = {string}")
+    value = parse_declare_value(chars_dir, Types.string, 8)
+    if value.__class__ is not str:
+        raise ValueError("Type mismatch: expected {0} got {1}".format(str, value.__class__))
+    var_name = parse_string_value([name_directory])
+    attach_variable(var_name, value)
+    print(f"VAR TYPE STRING {var_name} = {value}")
 
 
-def parse_string_value(chars_dir):
-    string_array = []
-    for char_dir in chars_dir.get_directory_children():
-        if char_dir.dirlen() != 8:
+def declare_boolean(bit_dir, name_directory):
+    value = parse_declare_value(bit_dir, Types.boolean, 1)
+    if value.__class__ is not bool:
+        raise ValueError("Type mismatch: expected {0} got {1}".format(bool, value.__class__))
+    var_name = parse_string_value([name_directory])
+    attach_variable(var_name, value)
+    print(f"VAR TYPE STRING {var_name} = {value}")
+
+
+def parse_declare_value(data_dir, type_class, basic_type_length):
+    print(f"VAR TYPES Parsing declare value of type {type_class}")
+
+    parsing_function = parsing_dict[type_class]
+
```

Comment:
Update, operation requires in first subdir 2 dirs: 1. dir type - arithm, compare, string operations, and 2 dir value in enum